### ISSUE TITLE
amazon account with multiple security groups called "docker-machine" may fail to create instance

### ISSUE THREAD

**Issue by jeffellin** [docker/machine#2810]
AWS can contain multiple security groups with the name "docker-machine" They may exist in multiple VPCs.   The driver currently requires subnet as a minimum input,  however vpc is not required.  Without the vpc information the driver may retrieve the wrong security group resulting in an issue where the group cannot be applied to an instance in the specified subnet.

Suggestion,  

It should be possible to lookup to the vpcid when not supplied using the subnet-id. If that is not practical than vpc should be required as well.

**Comment by jeanlaurent** [comment]
@jeffellin since we removed the subnet/vpc_id as a mandatory field with #2828 this may have changed. 

If I understand what you suggest : should we ensure to look for a `docker-machine` security group belonging to the active VPC_ID ? Is that right ?

**Comment by jeffellin** [comment]
Yes,  when you query for docker-machine group you should add the vpc_id so it finds the right one.

**Comment by savagegus** [comment]
:+1:

**Comment by bahchis** [comment]
I am experiencing related issue with 0.5.6 version.

When specifying subnet id only without vpc id, if a group with the same name already exists the machine creation fails for duplicate security group error.

After debugging it a bit, I think the reason is that if vpc id is not specified (specifying it fixes the issue), the check for existing security group fails to fetch the group and docker machine continues to creating the group, although it may already exist. The failure is here: https://github.com/docker/machine/blob/master/drivers/amazonec2/amazonec2.go#L885 but the problem is probably above that line where the group is not found correctly.
I did not see this handled in #2828 (looked very briefly).

I would suggest the following fallback:
provided vpc-id -> vpc-id from subnet -> default vpc-id.

Does it make sense?
